### PR TITLE
Add namespace selector for create batch spec execution page

### PR DESCRIPTION
**Pull request by eseliger**
Selecting a namespace for execution can now be done through the API and UI.

image

Co-authored-by: Kelli Rockwell <kelli@sourcegraph.com>

### CODE REVIEW

File: cmd/frontend/graphqlbackend/batches.graphql [hunk 1]
```diff
@@ -2168,8 +2168,11 @@ extend type Mutation {
     Creates a new batch spec execution from a given batch spec yaml file input.
     The execution will be queued for processing by an executor. If some are available
     for work, they will pick this up eventually.
+
+
+    If namespace is not specified, the current users personal namespace is used.
```
Comment (courier-new) [inline cmd/frontend/graphqlbackend/batches.graphql:2173]:
```suggestion
    If namespace is not specified, the current user's personal namespace is used.
```

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx [hunk 2]
```diff
@@ -160,42 +152,115 @@ export const CreateBatchChangePage: React.FunctionComponent<CreateBatchChangePag
...
+        []
+    const userNamespace = namespaces.find(
+        (namespace): namespace is SettingsUserSubject => namespace.__typename === 'User'
+    )
+    const organizationNamespaces = namespaces.filter(
+        (namespace): namespace is SettingsOrgSubject => namespace.__typename === 'Org'
+    )
+
+    if (!userNamespace) {
+        throw new Error('Bye')
```
Comment (courier-new) [inline client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:188]: Oh, we forgot to give this an actual error message. 😂

**Review by courier-new — APPROVED** [review]

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx (hunk 2 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:188]: Whoups 😆

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx [hunk 3]
```diff
@@ -160,42 +152,115 @@ export const CreateBatchChangePage: React.FunctionComponent<CreateBatchChangePag
...
+                    className="mb-3"
+                    onChange={setContent}
+                />
+                <button
+                    type="button"
+                    className="btn btn-primary"
+                    onClick={submitBatchSpec}
+                    disabled={isLoading === true}
+                >
+                    Run batch spec
```
Comment (mrnugget) [inline client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:248]:
```suggestion
                    Execute batch spec
```

File: enterprise/internal/batches/resolvers/resolver.go [hunk 4]
```diff
@@ -1462,9 +1462,24 @@ func (r *Resolver) CreateBatchSpecExecution(ctx context.Context, args *graphqlba
...
 
 	exec := &btypes.BatchSpecExecution{
-		BatchSpec:       args.Spec,
-		UserID:          actor.UID,
-		NamespaceUserID: actor.UID,
+		BatchSpec: args.Spec,
+		UserID:    actor.UID,
+	}
+
+	// TODO: Add a test to verify namespace is correctly set and permissions are enforced properly.
```
Comment (mrnugget) [inline enterprise/internal/batches/resolvers/resolver.go:1469]: What's the plan for this TODO? Should be relatively easy to test this, if I'm not missing anything. Or do you want to address this in a follow-up?

**Review by mrnugget — APPROVED** [review]
Really cool!

(Also made me think that if we pull this out into a single page we could already start thinking about providing a preview of how many repositories the `repositoriesMatchingQuery` would yield)

File: enterprise/internal/batches/resolvers/resolver.go (hunk 4 above)
Comment (eseliger) [reply enterprise/internal/batches/resolvers/resolver.go:1469]: Ah sorry I wrote it in the PR description and not as a comment:

> I'll add the test later but the rest is ready for review.

I want to do this before merging.

**Pull request merged**